### PR TITLE
RDM-4687 - fix show hide for complex fields

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4687

### Change description ###
Fix for show and hide bug on complex fields (collection of complex fields, complex fields, nested complex etc..)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/directives/conditional-show/conditional-show.directive.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export class ConditionalShowDirective implements AfterViewInit, OnDestroy {
   @Input() contextFields: CaseField[] = [];
   @Input() formGroup: FormGroup;
   @Input() greyBarEnabled = false;
+  @Input() formSubGroup: FormGroup;
```
Comment (gbenadikar) [inline src/shared/directives/conditional-show/conditional-show.directive.ts]: introduced form sub group to keep the impact of the changes minimum

File: src/shared/directives/conditional-show/conditional-show.directive.ts [hunk 2]
```diff
@@ -41,7 +42,8 @@ export class ConditionalShowDirective implements AfterViewInit, OnDestroy {
       this.condition = new ShowCondition(this.caseField.show_condition);
       // console.log('FIELD: ' + this.caseField.id + ' init. Show condition: ' + this.caseField.show_condition);
       this.formGroup = this.formGroup || new FormGroup({});
-      this.formField = this.formGroup.get(this.caseField.id);
+      this.formSubGroup = this.formSubGroup || new FormGroup({});
+      this.formField = this.formSubGroup.get(this.caseField.id) || this.formGroup.get(this.caseField.id);
```
Comment (gbenadikar) [inline src/shared/directives/conditional-show/conditional-show.directive.ts]: please tale a closer look at this line 🙂. I lookup in the **formSubGroup** first and then **formGroup** as the case field id may be same for a root case field or a complex subfield.

File: src/shared/components/palette/complex/write-complex-field.html [hunk 3]
```diff
@@ -22,6 +22,7 @@ <h2 *ngIf="renderLabel" class="heading-h2">{{caseField | ccdFieldLabel}}</h2>
                        [contextFields]="caseFields"
                        [caseFields]="caseFields"
                        [formGroup]="formGroup"
+                       [formSubGroup]="complexGroup"
```
Comment (mario-paniccia) [inline src/shared/components/palette/complex/write-complex-field.html]: Giri shall we call it `complexForm` rather than `formSubGroup`?

Comment (gbenadikar) [reply src/shared/components/palette/complex/write-complex-field.html]: May be complexFormGroup?

Comment (mario-paniccia) [reply src/shared/components/palette/complex/write-complex-field.html]: ok :)

**Pull request merged**